proposals/0010-diet-clients.txt

```diff
@@ -0,0 +1,183 @@
+---
+simd: '0010'
+title: Diet Clients
+authors:
+  - Anatoly Yakovenko (Solana Labs)
+category: Sealevel Virtual Machine
+type: Standards Track
+status: Draft
+created: 2022-12-09
+---
+
+## Summary
+
+Users that don't run a full node need secure access to the solana
+blockchain. The goal of this proposal is to define a protocol for
+sampling the validators such that if at least a small minority of
+validators are not faulty the user can be notified of an invalid
+state transition or a double confirmation attack on the network.
+
+## Specification
+
+### Overview
+
+1. User observes a confirmation for a TX at slot N. Confirmation
+contains 2/3+ votes for slot N, and a path from the transaction to
+the bankhash signed by all the votes.
+
+2. Concurrently with 1, user requests votes for slot N from M random
+validators.  Votes returned may be for later slots, and would contain
+a merkle path to the requested slot's bankhash.
```

Comment:
If the slot's bankhash is the same across the requests, the correctly-verifiable merkle paths will also be the same (since it is computationally impossible to create two different merkle-paths leading to the same root-hash). Perhaps to reduce redundancy, you might NOT want to ask all of M validators to send merkle paths, since one path would be enough.